sdk/data/azappconfig/options.go

```diff
@@ -76,3 +76,127 @@ type SetSettingOptions struct {
 	// if the passed-in ETag is the same version as the one in the configuration store.
 	OnlyIfUnchanged *azcore.ETag
 }
+
+// BeginCreateSnapshotOptions contains the optional parameters for the BeginCreateSnapshot method.
+type BeginCreateSnapshotOptions struct {
+	// Resumes the LRO from the provided token.
+	ResumeToken string
+
+	// The composition type describes how the key-values within the snapshot are composed. The 'key' composition type ensures
+	// there are no two key-values containing the same key. The 'key_label' composition
+	// type ensures there are no two key-values containing the same key and label.
+	CompositionType *CompositionType
+
+	// The amount of time, in seconds, that a snapshot will remain in the archived state before expiring. This property is only
+	// writable during the creation of a snapshot. If not specified, the default
+	// lifetime of key-value revisions will be used.
+	RetentionPeriod *int64
+
+	// The tags of the snapshot.
+	Tags map[string]*string
+
+	// READ-ONLY; The time that the snapshot was created.
```

Comment:
I think all of these R/O fields can be removed as they're output only yes?

Reply:
yes I agree